components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/core/rule/PreUpdatePasswordActionRuleEvaluationDataProvider.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.pre.update.password.action.core.rule;
+
+import org.wso2.carbon.identity.core.context.IdentityContext;
+import org.wso2.carbon.identity.core.context.model.Flow;
+import org.wso2.carbon.identity.rule.evaluation.exception.RuleEvaluationDataProviderException;
+import org.wso2.carbon.identity.rule.evaluation.model.Field;
+import org.wso2.carbon.identity.rule.evaluation.model.FieldValue;
+import org.wso2.carbon.identity.rule.evaluation.model.FlowContext;
+import org.wso2.carbon.identity.rule.evaluation.model.FlowType;
+import org.wso2.carbon.identity.rule.evaluation.model.RuleEvaluationContext;
+import org.wso2.carbon.identity.rule.evaluation.model.ValueType;
+import org.wso2.carbon.identity.rule.evaluation.provider.RuleEvaluationDataProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Rule evaluation data provider for pre update password flow.
+ * This class provides the data required for rule evaluation in pre update password flow.
+ */
+public class PreUpdatePasswordActionRuleEvaluationDataProvider implements RuleEvaluationDataProvider {
+
+    /**
+     * Supported flows for pre update password action.
+     */
+    private enum PasswordUpdateFlowType {
+        ADMIN_INITIATED_PASSWORD_RESET("admin_initiated_password_reset"),
+        ADMIN_INITIATED_PASSWORD_UPDATE("admin_initiated_password_update"),
+        ADMIN_INITIATED_USER_INVITE_TO_SET_PASSWORD("admin_initiated_user_invite_to_set_password"),
+        APPLICATION_INITIATED_PASSWORD_UPDATE("application_initiated_password_update"),
+        USER_INITIATED_PASSWORD_UPDATE("user_initiated_password_update"),
+        USER_INITIATED_PASSWORD_RESET("user_initiated_password_reset");
+
+        final String flowName;
+
+        PasswordUpdateFlowType(String flowName) {
+
+            this.flowName = flowName;
+        }
+
+        public String getFlowName() {
+
+            return flowName;
+        }
+    }
+
+    private enum RuleField {
+        FLOW("flow");
+
+        final String fieldName;
+
+        RuleField(String fieldName) {
+
+            this.fieldName = fieldName;
+        }
+
+        public String getFieldName() {
+
+            return fieldName;
+        }
+
+        public static RuleField valueOfFieldName(String fieldName) throws RuleEvaluationDataProviderException {
+
+            for (RuleField ruleField : RuleField.values()) {
+                if (ruleField.getFieldName().equals(fieldName)) {
+                    return ruleField;
+                }
+            }
+
+            throw new RuleEvaluationDataProviderException("Unsupported field: " + fieldName);
+        }
+    }
```

Comment:
Is this class is really needed? This just have one field. So can't we have some simpler way

Reply:
Like just have a string constant for the 'flow'